nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -0,0 +1,409 @@
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        try {
+
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(InputStream inputStream) throws IOException {
+
+                    try {
+                        String desiredSheetsDelimited = context.getProperty(DESIRED_SHEETS)
+                                .evaluateAttributeExpressions().getValue();
+
+                        OPCPackage pkg = OPCPackage.open(inputStream);
+                        XSSFReader r = new XSSFReader(pkg);
+                        SharedStringsTable sst = r.getSharedStringsTable();
+                        XSSFReader.SheetIterator iter = (XSSFReader.SheetIterator) r.getSheetsData();
+
+                        if (desiredSheetsDelimited != null) {
+
+                            String[] desiredSheets = StringUtils
+                                    .split(desiredSheetsDelimited, DESIRED_SHEETS_DELIMITER);
+
+                            if (desiredSheets != null) {
+
+                                while (iter.hasNext()) {
+                                    InputStream sheet = iter.next();
+                                    String sheetName = iter.getSheetName();
+
+                                    for (int i = 0; i < desiredSheets.length; i++) {
+                                        //If the sheetName is a desired one parse it
+                                        if (sheetName.equalsIgnoreCase(desiredSheets[i])) {
+                                            handleExcelSheet(session, flowFile, sst, sheet, sheetName);
+                                            break;
+                                        }
+                                    }
+                                }
+                            } else {
+                                getLogger().debug("Excel document was parsed but no sheets with the specified desired names were found.");
+                            }
+
+                        } else {
+                            //Get all of the sheets in the document.
+                            while (iter.hasNext()) {
+                                handleExcelSheet(session, flowFile, sst, iter.next(), iter.getSheetName());
+                            }
+                        }
+                    } catch (Exception ex) {
+                        getLogger().error(ex.getMessage());
+                    }
+                }
+            });
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
```

Comment:
Please include a comment and the full exception trace, like `getLogger().error("Attempt to do X failed", ex)`.